utils/mil_ros_tools/online_bagger/nodes/online_bagger.py

```diff
@@ -0,0 +1,427 @@
+
+        self.subscribe_loop()
+        rospy.loginfo('subscriber success: %s', self.subscriber_list)
+
+    def get_params(self):
+
+        """
+        Retrieve parameters from param server.
+        """
+
+        self.topics = rospy.get_param('/online_bagger/topics')
+        self.ram_limit = rospy.get_param('/online_bagger/ram_limit') * 1.073e9  # gigabytes to bytes
+        # self.stream_time = rospy.get_param('/online_bagger/stream_time')
+        self.dir = rospy.get_param('/online_bagger/bag_package_path')
+        self.bag_name = rospy.get_param('/online_bagger/bag_name')
+
+        if not self.bag_name:
+            self.bag_name = str(datetime.date.today()) + '-' + str(datetime.datetime.now().time())[0:8]
+
+        # rospy.loginfo('stream_time: %s seconds', self.stream_time)
+        rospy.loginfo('ram_limit: %f gb', rospy.get_param('/online_bagger/ram_limit'))
+
+    def make_dicts(self):
+
+        """
+        Make dictionary with sliceable deques() that will be filled with messages and time stamps.
+
+        Subscriber list contains all of the topics and their subscription status:
+        A True status for a given topic corresponds to a successful subscription
+        A False status indicates a failed subscription
+
+        For Example:
+        self.subscriber_list[0] = [['/odom', False], ['/absodom', True]]
+
+        Indicates that '/odom' has not been subscribed to, but '/absodom' has been subscribed to
+
+        self.topic_messages is a dictionary of deques containing a list of tuples.
+        Dictionary Keys contain topic names
+        Each value for each topic contains a deque
+        Each deque contains a list of tuples
+        Each tuple contains a message and its associated time stamp
+
+        For example:
+        '/odom' is  a potential topic name
+        self.topic_message['/odom']  is a deque
+        self.topic_message['/odom'][0]  is the oldest message available in the deque
+        and its time stamp if available
+        """
+
+        # Courtesy Kindall
+        # http://stackoverflow.com/questions/10003143/how-to-slice-a-deque
+        class sliceable_deque(deque):
+            def __getitem__(self, index):
+                if isinstance(index, slice):
+                    return type(self)(itertools.islice(self, index.start,
+                                                       index.stop, index.step))
+                return deque.__getitem__(self, index)
+
+        self.topic_messages = {}
+        self.subscriber_list = []
+
+
+        for topic in self.topics:
+            self.topic_messages[topic] = sliceable_deque(deque())
+            self.subscriber_list.append([topic, False])
+
+        rospy.loginfo('failed topics: %s', self.subscriber_list)
+
+    def subscribe_loop(self):
+
+        """
+        Continue to subscribe until at least one topic is successful
+        """
+
+        i = 0
+        while not self.subscribe():
```

Comment:
Will this ever run? `self.i` will always be > 0 as soon as one topic is subscribed to.